src/cordova/platform/addHelper.js

```diff
@@ -292,7 +292,9 @@ function getVersionFromConfigFile (platform, cfg) {
 function downloadPlatform (projectRoot, platform, version, opts) {
     var target = version ? (platform + '@' + version) : platform;
     return Q().then(function () {
-        if (opts.fetch) {
+        var platformPath = path.join(projectRoot, 'platforms', platform);
+        var platformAlreadyAdded = fs.existsSync(platformPath);
+        if ((opts.fetch) && (!platformAlreadyAdded)) {
```

Comment:
So what do we return if `platformAlreadyAdded` is true? I think we need to return something to make sure we get to line 333. 

In line 333, what is `libDir` if it does fetch?

Reply:
For instance, if I add the ios platform to my project, libDir = /Users/auso/cordova/tests/project_9/node_modules/cordova-ios

Reply:
in line 333, `libDir` is defined?

Reply:
Yes! These are my logs:
`Error: Platform ios already added.
libDir = /Users/auso/.cordova/lib/npm_cache/cordova-ios/4.5.4/package`

Reply:
is your manual test using `fetch:true`

Reply:
Lets wait and rebase this after we merge your commit removing no-fetch

Reply:
Just rebased!